### PR TITLE
fix(curriculum): update calorie counter step 77 to not allow extra addition

### LINKED ISSUE #55840: Calorie Counter (step 77) test should be more strict
### Describe the Issue

**Learn Form Validation by Building a Calorie Counter**
Step 77

Exercise:
`It is time to start preparing your calculations. Start by declaring a consumedCalories variable, and assign it the sum of breakfastCalories, lunchCalories, dinnerCalories, and snacksCalories (note that order matters for the tests). Be sure to do this after your if statement.`

**Test is _not limited_ to the correct elements. I was able to submit this answer:** 
`const consumedCalories = breakfastCalories + lunchCalories + dinnerCalories + snacksCalories + exerciseCalories + budgetCalories;`

Test is strict in that answer must contain the first four elements in a specific order, but it is not limited to only just that.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-form-validation-by-building-a-calorie-counter/step-77

### Your code

```
const consumedCalories = breakfastCalories + lunchCalories + dinnerCalories + snacksCalories + exerciseCalories + budgetCalories;

```

### Expected behavior

My answer should not pass the test checks, as it contains information that is incorrect. Test should be limited to first four elements.

### Screenshots

_No response_

### System

- Browser: Brave

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #55840

The best way I thought of to resolve the issue is to simply make sure the line of code ends where it is supposed to with a semi-colon.

### CODE REVIEW

**Comment by lasjorg** [comment]
You should be able to use `$` with the `m` flag.

---
````md
You should calculate the sum of `breakfastCalories`, `lunchCalories`, `dinnerCalories`, and `snacksCalories`, in order.

```js
assert.match(calculateCalories.toString(), /breakfastCalories\s*\+\s*lunchCalories\s*\+\s*dinnerCalories\s*\+\s*snacksCalories;?$/m);
```

You should assign the sum of `breakfastCalories`, `lunchCalories`, `dinnerCalories`, and `snacksCalories` to `consumedCalories`.

```js
assert.match(calculateCalories.toString(), /consumedCalories\s*=\s*breakfastCalories\s*\+\s*lunchCalories\s*\+\s*dinnerCalories\s*\+\s*snacksCalories;?$/m);
```
````

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-form-validation-by-building-a-calorie-counter/63c9e4d2ff41811dd640504f.md [hunk 1]
```diff
@@ -35,6 +35,12 @@ You should assign the sum of `breakfastCalories`, `lunchCalories`, `dinnerCalori
 assert.match(calculateCalories.toString(), /consumedCalories\s*=\s*breakfastCalories\s*\+\s*lunchCalories\s*\+\s*dinnerCalories\s*\+\s*snacksCalories/);
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-form-validation-by-building-a-calorie-counter/63c9e4d2ff41811dd640504f.md:35]:
```suggestion
assert.match(calculateCalories.toString(), /consumedCalories\s*=\s*breakfastCalories\s*\+\s*lunchCalories\s*\+\s*dinnerCalories\s*\+\s*snacksCalories;?$/m);
```

Comment (lasjorg) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-form-validation-by-building-a-calorie-counter/63c9e4d2ff41811dd640504f.md:35]: Might as well do the same here.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-form-validation-by-building-a-calorie-counter/63c9e4d2ff41811dd640504f.md [hunk 2]
```diff
@@ -35,6 +35,12 @@ You should assign the sum of `breakfastCalories`, `lunchCalories`, `dinnerCalori
...
 ```
 
+You should end the line of code with a semi-colon. 
+
+```js
+assert.match(calculateCalories.toString(), /consumedCalories\s*=\s*breakfastCalories\s*\+\s*lunchCalories\s*\+\s*dinnerCalories\s*\+\s*snacksCalories;?$/m);
+```
+
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-form-validation-by-building-a-calorie-counter/63c9e4d2ff41811dd640504f.md:43]:
```suggestion
```

The semicolon is optional `;?`

So we shouldn't need this test.

**Review by lasjorg — CHANGES_REQUESTED** [review]

**Review by lasjorg — APPROVED** [review]

**Comment by a2937** [comment]
Hey, regarding this https://github.com/freeCodeCamp/freeCodeCamp/issues/53694? 

Am I making this problem worse? I mean it's literally the only thing I could come up with at the time.

**Comment by lasjorg** [comment]
Not sure I understand the question.

The tests in this PR do not require semicolons. But if it had, that would have been a questionable solution. Semicolons should be optional, unless specifically chosen not to be for some educational reason.

**Comment by a2937** [comment]
I think I just got confused about what the issue was about. You see for campers to pass these tests, they're going to need to insert semi-colons as that's how we can safely cut off content. 

I thought the issue meant that we should either enforce campers to consistently use semi-colons or not be strict about it.

**Comment by lasjorg** [comment]
> You see for campers to pass these tests, they're going to need to insert semi-colons as that's how we can safely cut off content.

The tests in this PR do not rely on semicolons, they are optional.

---

The issue you linked to is about semicolons in general. How we teach them, when to require them, etc.

**Review by gikf — APPROVED** [review]

**Comment by lasjorg** [comment]
@gikf I was going to add something like that, but then I seemed to remember having tested it so I tested it again and the regex already supported the optional spaces.

In any case, having it explicit in the regex is probably a good idea. I just wanted to mention it because it caught me by surprise as well.

**Comment by gikf** [comment]
@lasjorg Do you mean with latest changes, or before?

**Comment by lasjorg** [comment]
Before.

spaces-oldregex

**Comment by gikf** [comment]
@lasjorg Ah! It's the `calculateCalories.toString()` that's cleaning it up.

**Pull request merged**